### ISSUE TITLE
Getting error Proj version 0.0.0 is installed, but cartopy requires at least version 4.9.0 during the installation on hosted server on godaddy website.

### ISSUE THREAD

**Issue by krishnaap** [SciTools/cartopy#1793]
I was trying to build a web app by hosting my codes on a GoDaddy hosting server. I tried to install cartopy using `pip install cartopy` on my python version 3.7.8. I was asking to install dependencies such as Cython and Proj. I installed Cython using pip. Then installed Proj using pip. Then I tried to run `pip install cartopy, its showing the error Getting error Proj version 0.0.0 is installed, but cartopy requires at least version 4.9.0 

pip install cartopy
Collecting cartopy
  Using cached Cartopy-0.19.0.post1.tar.gz (12.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/yg882598du7o/virtualenv/rainlab/3.7/bin/python3.7_bin /home/yg882598du7o/virtualenv/rainlab/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp_6giij_q
       cwd: /tmp/pip-install-to3wey4m/cartopy_99d769978bd94e86abc0239fe04ff72b
  Complete output (47 lines):
  Traceback (most recent call last):
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/core/__init__.py", line 16, in <module>
      from . import multiarray
  ImportError: /tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/core/multiarray.cpython-37m-x86_64-linux-gnu.so: failed to map segment from shared object: Operation not permitted
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "setup.py", line 58, in <module>
      import numpy as np
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/__init__.py", line 142, in <module>
      from . import add_newdocs
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/add_newdocs.py", line 13, in <module>
      from numpy.lib import add_newdoc
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/lib/__init__.py", line 8, in <module>
      from .type_check import *
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/lib/type_check.py", line 11, in <module>
      import numpy.core.numeric as _nx
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/core/__init__.py", line 26, in <module>
      raise ImportError(msg)
  ImportError:
  Importing the multiarray numpy extension module failed.  Most
  likely you are trying to import a failed build of numpy.
  If you're working with a numpy git repo, try `git clean -xdf` (removes all
  files not under version control).  Otherwise reinstall numpy.
  
  Original error was: /tmp/pip-build-env-fe44ay7f/overlay/lib64/python3.7/site-packages/numpy/core/multiarray.cpython-37m-x86_64-linux-gnu.so: failed to map segment from shared object: Operation not permitted
  
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/home/yg882598du7o/virtualenv/rainlab/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 280, in <module>
      main()
    File "/home/yg882598du7o/virtualenv/rainlab/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/yg882598du7o/virtualenv/rainlab/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 155, in get_requires_for_build_wheel
      config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-fe44ay7f/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 60, in <module>
      raise ImportError('NumPy 1.10+ is required to install cartopy.')
  ImportError: NumPy 1.10+ is required to install cartopy.
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/ed/ca/524ce33692df3faeaa56852fb6a33b0b410be94cc288417565b96fef3f64/Cartopy-0.19.0.post1.tar.gz#sha256=4b8b4773a98ed7009fe17d9b6ec87ac3ac62b7d14634d7768c190eadc647d576 (from https://pypi.org/simple/cartopy/) (requires-python:>=3.5). Command errored out with exit status 1: /home/yg882598du7o/virtualenv/rainlab/3.7/bin/python3.7_bin /home/yg882598du7o/virtualenv/rainlab/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp_6giij_q Check the logs for full command output.
  Using cached Cartopy-0.18.0.tar.gz (14.4 MB)
    ERROR: Command errored out with exit status 1:
     command: /home/yg882598du7o/virtualenv/rainlab/3.7/bin/python3.7_bin -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-to3wey4m/cartopy_ae3167786a934868904a2c968adf457a/setup.py'"'"'; __file__='"'"'/tmp/pip-install-to3wey4m/cartopy_ae3167786a934868904a2c968adf457a/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-41clt4gl
         cwd: /tmp/pip-install-to3wey4m/cartopy_ae3167786a934868904a2c968adf457a/
    Complete output (5 lines):
    /tmp/pip-install-to3wey4m/cartopy_ae3167786a934868904a2c968adf457a/setup.py:107: UserWarning: Unable to determine GEOS version. Ensure you have 3.3.3 or later installed, or installation may fail.
      '.'.join(str(v) for v in GEOS_MIN_VERSION), ))
    /tmp/pip-install-to3wey4m/cartopy_ae3167786a934868904a2c968adf457a/setup.py:160: UserWarning: Unable to determine Proj version. Ensure you have 4.9.0 or later installed, or installation may fail.
      '.'.join(str(v) for v in PROJ_MIN_VERSION), ))
    Proj version 0.0.0 is installed, but cartopy requires at least version 4.9.0.
    ----------------------------------------

#### Code to reproduce

```

```

#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### conda list

```
```

### pip list

```
```

</details>

**Comment by dopplershift** [comment]
If you installed "Proj" using pip, I'm pretty sure that's only PyProj, not the C-library, or more importantly, the headers. On linux you'll need to install something like libproj-dev and libgeos-dev (depends on your distribution). Also, some of those errors also look like numpy is not installed.

**Comment by MosGeo** [comment]
Simplifying things for pip users would be great. I am not expert on Github actions but automating building a pip wheel similar to Christoph Gohlke unofficial binary and pushing it to pypi is the best option moving forward.
 
Christoph Gohlke (https://www.lfd.uci.edu/~gohlke/pythonlibs/) maintains unofficial Windows binaries of cartopy.

**Comment by krishnaap** [comment]
Sorry to bother you with this issue. I contacted GoDaddy technical teams about this, they told me, GoDaddy is restricting usage of Python version later than 2.6.6 on their shared hosting websites. So I suppose I tried to install Python 3.7 and the necessary packages aren't possible to install by their system restrictions. Thanks for your kind response and the time you focused on this issue. As a matter of fact, I am trying to move on to Google Cloud Platform or so to deploy my python codes. Thanks again.

> Simplifying things for pip users would be great. I am not expert on Github actions but automating building a pip wheel similar to Christoph Gohlke unofficial binary and pushing it to pypi is the best option moving forward.
> 
> Christoph Gohlke (https://www.lfd.uci.edu/~gohlke/pythonlibs/) maintains unofficial Windows binaries of cartopy.

I agree with you, in fact, while installing via conda, it installing necessary dependencies along with the installation. But here in pip install it will not do that. I would like to see the change in that feature.

Thanks @dopplershift and @MosGeo
